src/frontend/src/components/Booking/Booking.tsx

```diff
@@ -0,0 +1,348 @@
+"use client";
+import customerBookingService from "@/services/customerBooking";
+import photographerBookingService from "@/services/photographerBooking";
+import { Booking, BookingStatus } from "@/types/booking";
+import { useEffect, useState } from "react";
+import BookingCard from "./BookingCard";
+import BookingModal from "./BookingModal";
+import { list } from "postcss";
+
+const mockData:Booking={
+  "id": "6ae48b2e-e6a3-43a0-8c69-cb677d56c257",
+  "customer_id": "3099fe61-ce94-4689-9553-45ef54ff511f",
+  "gallery": {
+      "id": "a0598881-cea7-48d0-90d0-ee7c0aa5fa27",
+      "photographer_id": "1ea3b074-61c4-4ca5-8e9a-0942f09111d8",
+      "location": "Chula",
+      "name": "Test Gallery",
+      "price": 6000,
+      "hours": 4,
+      "description": null,
+      "delivery_time": 10,
+      "included": []
+  },
+  "start_time": "2024-01-20T09:00:00Z",
+  "end_time": "2024-01-20T09:20:00Z",
+  "status": BookingStatus.BookingPaidOutStatus,
+  "created_at": "2024-03-22T13:04:18.277234Z",
+  "updated_at": "2024-03-22T13:04:18.277234Z"
+}
+
+
+export default function BookingPage() {
+  //----------Instance Var------------
+
+  const [bookingLists, setBookingLists] = useState<Booking[]|null>([]);
+  const [allAppointment, setAllAppointment] = useState<Boolean>(true);
+  const [upComing, setUpComing] = useState<Boolean>(false);
+  const [past, setPast] = useState<Boolean>(false);
+  const [cancel, setCancel] = useState<Boolean>(false);
+
+  //-------Fetch Object will be replaced later--------------------------------------
+  const getAllBookings = async () => {
+    let resultCustomer:Array<Booking>;
+    let resultPhotographer:Array<Booking>;
+  
+    try{
+      const res= await customerBookingService.getAllBookings();
+      if(res.data){
+        resultCustomer=res.data;
+      }
+      else{
+        resultCustomer=[]
+      }
+    }
+    catch(err){
+      resultCustomer=[];
```

Comment:
Please use the existing `useErrorModal` to display errors before setting the data to empty, in order to align with web best practices. You can find an example in /components/Gallery/MyGallery

Reply:
Currently the component is galleryInfo kub